models/ast/node_evaluation_dto.go

```diff
@@ -3,17 +3,31 @@ package ast
 import "github.com/checkmarble/marble-backend/pure_utils"
 
 type NodeEvaluationDto struct {
+	// When too long, the ReturnValue can be omitted (ex: when function is `FUNC_CUSTOM_LIST_ACCESS``).
+	// In such cases, ReturnValue will be the constant `omittedReturnValue``
 	ReturnValue   any                          `json:"return_value"`
 	Errors        []EvaluationErrorDto         `json:"errors"`
 	Children      []NodeEvaluationDto          `json:"children,omitempty"`
 	NamedChildren map[string]NodeEvaluationDto `json:"named_children,omitempty"`
 }
 
 func AdaptNodeEvaluationDto(evaluation NodeEvaluation) NodeEvaluationDto {
-	return NodeEvaluationDto{
+	nodeEvaluationDto := NodeEvaluationDto{
 		ReturnValue:   evaluation.ReturnValue,
 		Errors:        pure_utils.Map(evaluation.Errors, AdaptEvaluationErrorDto),
 		Children:      pure_utils.Map(evaluation.Children, AdaptNodeEvaluationDto),
 		NamedChildren: pure_utils.MapValues(evaluation.NamedChildren, AdaptNodeEvaluationDto),
 	}
+
+	if isReturnValueOmitted(evaluation) {
+		nodeEvaluationDto.ReturnValue = omittedReturnValue
```

Comment:
How about storing an empty array (or maybe the N first values?) rather than an arbitrary constant string (in the case of list access) ?

Reply:
I plan to store the arbitrary constant for every ommitted return values (so I can easilly parce the result frontend side and explicitly know when the result is skipped).

I know this is not elegant, but with the current form I can't get the information w\ adding another prop to the DTO.

Reply:
hmm ok, and it needs to be something else than NULL right

Reply:
fine then
maybe use an all caps string constant but that's just a matter of taste

Reply:
I'm afraid I don't have better either and we can always change this storage later if needed

Reply:
An alternative would be to :
- store `null`
- store a `boolean` prop like `isResultOmmitted`(or alike)